pyomo/contrib/parmest/parmest.py

```diff
@@ -1726,10 +1891,65 @@ def _Q_opt(
                 the constant cancels out. (was scaled by 1/n because it computes an
                 expected value.)
                 '''
-                cov = 2 * sse / (n - l) * inv_red_hes
-                cov = pd.DataFrame(
-                    cov, index=thetavals.keys(), columns=thetavals.keys()
-                )
+                if self.obj_function == 'SSE': # covariance calculation for measurements in the same unit
```

Comment:
Are you adding the same code in two places? If so, can we streamline this into a function to avoid duplicate code?

Reply:
Suggestion: remove this. Do not add this capability for the depreciated interface.

Reply:
You can even add an error message that computing the covariance is no longer supported in the deprecated interface.